tests/test_itk_torch_bridge.py

```diff
@@ -0,0 +1,525 @@
+        itk_transform.SetCenter(get_itk_image_center(image))
+
+    # Set matrix and translation
+    itk_transform.SetMatrix(itk.matrix_from_array(matrix))
+    itk_transform.Translate(translation)
+
+    # Interpolator
+    image = image.astype(itk.D)
+    interpolator = itk.LinearInterpolateImageFunction.New(image)
+
+    if not reference_image:
+        reference_image = image
+
+    # Resample with ITK
+    output_image = itk.resample_image_filter(
+        image, interpolator=interpolator, transform=itk_transform, output_parameters_from_image=reference_image
+    )
+
+    return np.asarray(output_image, dtype=np.float32)
+
+
+def monai_affine_resample(metatensor, affine_matrix):
+    affine = Affine(affine=affine_matrix, padding_mode="zeros", mode="bilinear", dtype=torch.float64, image_only=True)
+    output_tensor = affine(metatensor)
+
+    return output_tensor.squeeze().permute(*torch.arange(output_tensor.ndim - 2, -1, -1)).array
+
+
+def remove_border(image):
+    """
+    MONAI seems to have different behavior in the borders of the image than ITK.
+    This helper function sets the border of the ITK image as 0 (padding but keeping
+    the same image size) in order to allow numerical comparison between the
+    result from resampling with ITK/Elastix and resampling with MONAI.
+    To use: image[:] = remove_border(image)
+    Args:
+        image: The ITK image to be padded.
+
+    Returns:
+        The padded array of data.
+    """
+    return np.pad(image[1:-1, 1:-1, 1:-1] if image.ndim == 3 else image[1:-1, 1:-1], pad_width=1)
+
+
+def itk_warp(image, ddf):
+    """
+    warping with python itk
+    Args:
+        image: itk image of array shape 2D: (H, W) or 3D: (D, H, W)
+        ddf: numpy array of shape 2D: (2, H, W) or 3D: (3, D, H, W)
+    Returns:
+        warped_image: numpy array of shape (H, W) or (D, H, W)
+    """
+    # MONAI -> ITK ddf
+    displacement_field = monai_to_itk_ddf(image, ddf)
+
+    # Resample using the ddf
+    interpolator = itk.LinearInterpolateImageFunction.New(image)
+    warped_image = itk.warp_image_filter(
+        image, interpolator=interpolator, displacement_field=displacement_field, output_parameters_from_image=image
+    )
+
+    return np.asarray(warped_image)
+
+
+def monai_warp(image_tensor, ddf_tensor):
+    """
+    warping with MONAI
+    Args:
+        image_tensor: torch tensor of shape 2D: (1, 1, H, W) and 3D: (1, 1, D, H, W)
+        ddf_tensor: torch tensor of shape 2D: (1, 2, H, W) and 3D: (1, 3, D, H, W)
+    Returns:
+        warped_image: numpy array of shape (H, W) or (D, H, W)
+    """
+    warp = Warp(mode="bilinear", padding_mode="zeros")
+    warped_image = warp(image_tensor.to(torch.float64), ddf_tensor.to(torch.float64))
+
+    return warped_image.to(torch.float32).squeeze().numpy()
+
+
+@unittest.skipUnless(has_itk, "Requires `itk` package.")
+class TestITKTorchAffineMatrixBridge(unittest.TestCase):
+    def setUp(self):
+        torch.manual_seed(RANDOM_SEED)
```

Comment:
please consider these setup and teardown to avoid changing the global random state:

https://github.com/Project-MONAI/MONAI/blob/58c256eae2f1d21436a0b9c0dab5d989b8665e55/tests/test_inverse.py#L367-L371

https://github.com/Project-MONAI/MONAI/blob/58c256eae2f1d21436a0b9c0dab5d989b8665e55/tests/test_inverse.py#L395-L396